internal/hmyapi/apiv1/blockchain.go

```diff
@@ -260,61 +264,78 @@ func (s *PublicBlockChainAPI) GetBlockSigners(ctx context.Context, blockNr rpc.B
 	}
 	mask, err := internal_bls.NewMask(pubkeys, nil)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	err = mask.SetMask(blockWithSigners.Header().LastCommitBitmap())
 	if err != nil {
+		return nil, nil, err
+	}
+	return committee.Slots, mask, nil
+}
+
+// GetBlockSigners returns signers for a particular block.
+func (s *PublicBlockChainAPI) GetBlockSigners(ctx context.Context, blockNr rpc.BlockNumber) ([]string, error) {
+	if uint64(blockNr) == 0 {
+		return []string{}, nil
+	}
+	if err := s.isBlockGreaterThanLatest(blockNr); err != nil {
```

Comment:
move this inside `getBlockSigners` as no behavior is changed.

Reply:
Moved to the API backend, will keep the check in each function, because the checks are different for V1 & V2.